pallets/schemas/src/lib.rs

```diff
@@ -381,11 +445,204 @@ pub mod pallet {
 		) -> DispatchResult {
 			let sender = ensure_signed(origin)?;
 
-			let schema_id = Self::create_schema_for(model, model_type, payload_location, settings)?;
+			let (schema_id, _) =
+				Self::create_schema_for(model, model_type, payload_location, settings, None)?;
+
+			Self::deposit_event(Event::SchemaCreated { key: sender, schema_id });
+			Ok(())
+		}
+
+		/// Propose to create a schema.  Creates a proposal for council approval to create a schema
+		///
+		#[pallet::call_index(5)]
+		#[pallet::weight(
+			match schema_name {
+				Some(_) => T::WeightInfo::propose_to_create_schema_v2(model.len() as u32),
+				None => T::WeightInfo::propose_to_create_schema(model.len() as u32)
+			}
+		)]
+		pub fn propose_to_create_schema_v2(
+			origin: OriginFor<T>,
+			model: BoundedVec<u8, T::SchemaModelMaxBytesBoundedVecLimit>,
+			model_type: ModelType,
+			payload_location: PayloadLocation,
+			settings: BoundedVec<SchemaSetting, T::MaxSchemaSettingsPerSchema>,
+			schema_name: Option<SchemaNamePayload>,
+		) -> DispatchResult {
+			let proposer = ensure_signed(origin)?;
+
+			let proposal: Box<T::Proposal> = Box::new(
+				(Call::<T>::create_schema_via_governance_v2 {
+					creator_key: proposer.clone(),
+					model,
+					model_type,
+					payload_location,
+					settings,
+					schema_name,
+				})
+				.into(),
+			);
+			T::ProposalProvider::propose_with_simple_majority(proposer, proposal)?;
+			Ok(())
+		}
+
+		/// Create a schema by means of council approval
+		///
+		/// # Events
+		/// * [`Event::SchemaCreated`]
+		///
+		/// # Errors
```

Comment:
Lots of new events and errors due to the name updates.

The list of errors is getting long, so not sure we keep that correctly up to date.

Reply:
Thanks for the reminder. Updated the docs.